### PR TITLE
Replace LinkedFiles backslashes with forward slashes

### LINKED ISSUE #3311: Filepath saved on Windows does not open on Linux
```
JabRef 4.0
Linux 4.10.0-37-generic amd64 
Java 1.8.0_144
```

Steps to reproduce:

1. create an idential folderstructure on Windows and on Linux (f.e. use sync-systems like: dropbox, onedrive, googledrive, owncloud)
2. JabRef on Windows&Linux must have the same "Main file directory" (f.e. Linux: /home/jkalliau/ownCloud and Windows: C:/Users/jkalliau/ownCloud)
3. add a PDF-File to an article in JabRef to a bibtex-file on Windows
4. try to open this PDF-File in JabRef on Linux

The problem is that Windows saves:
```
File = {:..\\..\\Literature\\DNA\\Weitere\\Bai_Martin_Scheres_Dietz_2012_PNAS.pdf:PDF},
file = {:PhDJK\\Literature\\MD\\Amber17.pdf:PDF},
```
But Linux saves as:
```
file = {:PhDJK/Literature/MD/Amber17.pdf:PDF},
```
With JabRef 3.8.2 neither Windows nor Linux had problems using different File-Structures, but now with JabRef 4 on Linux it does (JabRef can't find the file). :-(

On Windows with JabRef4.0 it still works also with Linux-Filepathes.

This bib-file has developed over some time with JabRef 3.x and it always worked to open with JabRef 3.8.2 on Windows and on Linux, also both added file-paths in a different form.

I would stay with JabRef 3.8.2, but due to [bug #2879](https://github.com/JabRef/jabref/issues/2879) I prefer JabRef 4.

Otherwise I might have to use [sed](https://en.wikipedia.org/wiki/Sed) to correct the path-mistakes.

<details>
  <summary>bib-file with filepath saved on Windows</summary>
    ```
% Encoding: UTF-8


@Article{bai2012cryo,
  Title                    = {{Cryo-EM} structure of a {3D} {DNA-origami} object},
  Author                   = {Bai, Xiao-chen and Martin, Thomas~G and Scheres, Sjors~HW and Dietz, Hendrik},
  Journal                  = {Proceedings of the National Academy of Sciences},
  Year                     = {2012},

  Month                    = dec,
  Number                   = {49},
  Pages                    = {20012--20017},
  Volume                   = {109},

  Adsnote                  = {Provided by the SAO/NASA Astrophysics Data System},
  Adsurl                   = {http://adsabs.harvard.edu/abs/2012PNAS..10920012B },
  Doi                      = {10.1073/pnas.1215713109},
  File                     = {:..\\..\\Literature\\DNA\\Weitere\\Bai_Martin_Scheres_Dietz_2012_PNAS.pdf:PDF},
  Groups                   = {DNA},
  Keywords                 = {rank2},
  Publisher                = {National Acad Sciences}
}

@Article{case2017amber,
  author  = {Case, David A and Walker, Ross C and Cheatham III, Thomas E and Simmerling, Carlos LI and Roitberg, Adrian and Merz, Kenneth M and Luo, Ray and Darden, Tom A and Wang, Junmei and Duke, Robert E and Daniel R. Roe and Scott LeGrand and Jason Swails and Andreas W. Götz and David Cerutti and Tyler Luchko and Gérald Monard and Celeste Sagui and Feng Pan and Charles Lin and Daniel Mermelstein and Pengfei Li and Alexey Onufriev and Saeed Izadi and Romain M. Wolf and Xiongwu Wu and Holger Gohlke and Nadine Homeyer and Ruxi Qi and Wesley M. Botello-Smith and Li Xiao and D’Artagnan Greene and Tim Giese and Taisung Lee and Darrin York and Jian Liu and Hai Nguyen and Igor Omelyan and Andriy Kovalenko and Peter A. Kollman and Zhang, W and others},
  title   = {Amber 2017 Reference Manual (Covers Amber16 and AmberTools17)},
  journal = {University of California, San Francisco},
  year    = {2017},
  month   = may,
  day     = {29},
  file    = {:PhDJK\\Literature\\MD\\Amber17.pdf:PDF},
  url     = {http://ambermd.org/doc12/Amber17.pdf},
}

@Comment{jabref-meta: databaseType:bibtex;}

@Comment{jabref-meta: grouping:
0 AllEntriesGroup:;
1 StaticGroup:DNA\;0\;1\;\;\;\;;
}
  ```
</details>

### PR DESCRIPTION
**Pull request by Siedlerchr**
Ensures cross platform compatibility
Fixes #3311 

Tested this under windows and it works fine

- [ ] Change in CHANGELOG.md described
- [ ] Tests created for changes
- [ ] Screenshots added (for bigger UI changes)
- [x] Manually tested changed features in running JabRef
- [ ] Check documentation status (Issue created for outdated help page at [help.jabref.org](https://github.com/JabRef/help.jabref.org/issues)?)
- [ ] If you changed the localization: Did you run `gradle localizationUpdate`?

### CODE REVIEW

File: src/main/java/org/jabref/model/util/FileHelper.java [hunk 1]
```diff
@@ -15,6 +15,15 @@
 public class FileHelper {
 
     /**
+     * Checks if the given String is an online link
+     * @param toCheck The String to check
+     * @return True if it starts with http://, https:// or contains www; false otherwise
+     */
+    public static boolean isOnlineLink(String toCheck) {
+        return toCheck.startsWith("http://") || toCheck.startsWith("https://") || toCheck.contains("www.");
```
Comment (lynyus) [inline src/main/java/org/jabref/model/util/FileHelper.java]: URL validation is quite hard (potentially completely infeasible) in practice.

Is this implementation sufficient for our purposes? I mean "jabref.org" is a valid url, but would `isOnlineLink()` would return false.

My thoughts are to leave this method in the `LinkedFile` if it should serve as a "special purpose" link checker. If it should be a general method, in the `util` package, then the implementation would need to become more elaborate.

Comment (Siedlerchr) [reply src/main/java/org/jabref/model/util/FileHelper.java]: I did not change the detection of urls, I just moved it

Comment (LinusDietz) [reply src/main/java/org/jabref/model/util/FileHelper.java]: that's what I mean: if you move it to an `util` package, it should be a general purpose method that works in 'all' cases.

If it stays in this class (potentially private) it's sufficient if it is good enough to be called in that specific context.

**Comment by Siedlerchr** [comment]
yes, agree, good point  I readded it in the other PR

**Pull request closed without merging**